File: backend/src/main/java/ru/bowling/bowlingapp/DTO/SupplierComplaintRequestDTO.java
package ru.bowling.bowlingapp.DTO;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.bowling.bowlingapp.Entity.enums.SupplierComplaintStatus;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SupplierComplaintRequestDTO {
    @NotBlank
    private String title;

    @NotBlank
    private String description;

    @NotNull
    private SupplierComplaintStatus status;
}

File: frontend/lib/models/supplier_complaint_request_dto.dart
class SupplierComplaintRequestDto {
  final String title;
  final String description;
  final String status;

  SupplierComplaintRequestDto({
    required this.title,
    required this.description,
    required this.status,
  });

  Map<String, dynamic> toJson() => {
        'title': title,
        'description': description,
        'status': status,
      };
}
